crates/goose/src/context_mgmt/auto_compact.rs

```diff
@@ -300,7 +317,7 @@ mod tests {
         // Create small messages that won't trigger compaction
         let messages = vec![create_test_message("Hello"), create_test_message("World")];
 
-        let result = check_compaction_needed(&agent, &messages, Some(0.3))
+        let result = check_compaction_needed(&agent, &messages, Some(0.3), None)
```

Comment:
do we ever test the meta_data in these calls? I'd maybe kill a few of these tests and have that does